tools/generate_release_notes.py

```diff
@@ -1,246 +1,331 @@
+        return f"`{name} <{target}>`_"
 
-# add Other PRs to the ordered dict to make doc generation easier.
-highlights["Other Pull Request"] = other_pull_requests
+    def _format_section_title(self, title: str, level: int) -> Iterable[str]:
+        yield title + "\n"
+        underline = {1: "=", 2: "-", 3: "~"}
+        yield underline[level] * len(title) + "\n"
 
 
-# Now generate the release notes
-announcement_title = f"scikit-image {args.version} release notes"
-print(announcement_title)
-print("=" * len(announcement_title))
+def parse_command_line(func: Callable) -> Callable:
+    """Define and parse command line options.
 
-print(
+    Has no effect if any keyword argument is passed to the underlying function.
     """
-scikit-image is an image processing toolbox for SciPy that includes algorithms
-for segmentation, geometric transformations, color space manipulation,
-analysis, filtering, morphology, feature detection, and more.
-"""
-)
+    parser = argparse.ArgumentParser(usage=__doc__)
+    parser.add_argument("start_rev", help="The starting revision (excluded)")
+    parser.add_argument("stop_rev", help="The stop revision (included)")
+    parser.add_argument(
+        "--version", help="Version you're about to release", default="0.0.0"
+    )
+    parser.add_argument("--out", help="Write to file, prints to STDOUT otherwise")
+    parser.add_argument(
+        "--format",
+        choices=["rst", "md"],
+        default="md",
+        help="Choose format, defaults to Markdown",
+    )
 
-print(
-    """
-For more information, examples, and documentation, please visit our website:
+    def wrapped(**kwargs):
+        if not kwargs:
+            kwargs = vars(parser.parse_args())
+        return func(**kwargs)
 
-https://scikit-image.org
+    return wrapped
 
-"""
-)
-
-for section, pull_request_dicts in highlights.items():
-    if not pull_request_dicts:
-        continue
-    print(f'{section}s\n{"*" * (len(section)+1)}')
-    for pr_num, pr_info in pull_request_dicts.items():
-        print(
-            f"- {pr_info['summary'].strip()}\n  (`#{pr_num} <https://github.com/scikit-image/scikit-image/pull/{pr_num}>`_)."
-        )
 
-contributors = {}
+@parse_command_line
+def main(*, start_rev: str, stop_rev: str, version: str, out: str, format: str):
+    # TODO option to delete cache
+    requests_cache.install_cache("github_cache", backend="sqlite", expire_after=3600)
 
-contributors["authors"] = authors
-# contributors['committers'] = committers
-contributors["reviewers"] = reviewers
+    gh_token = os.environ.get("GH_TOKEN")
+    if gh_token is None:
+        raise RuntimeError(
+            "It is necessary that the environment variable `GH_TOKEN` "
+            "be set to avoid running into problems with rate limiting. "
+            "One can be acquired at https://github.com/settings/tokens.\n\n"
+            "You do not need to select any permission boxes while generating "
+            "the token."
```

Comment:
```suggestion
            "You need to set the environment variable `GH_TOKEN`. "
            "The token is used to avoid rate limiting, "
            "and can be created at https://github.com/settings/tokens.\n\n"
            "The token does not require any permissions (we only use the public API)."
```